### PR TITLE
for chatmail accounts, hide some not useful settings

### PR DESCRIPTION
**Pull request by r10s**
counterpart of https://github.com/deltachat/deltachat-android/pull/3050 ,   see there for discussion and reasoning

this is how the advanced settings look like now: left classicmail, right chatmail ... so much cleaner :)

### CODE REVIEW

File: deltachat-ios/Controller/Settings/AdvancedViewController.swift [hunk 1]
```diff
@@ -56,17 +56,17 @@ internal final class AdvancedViewController: UITableViewController, ProgressAler
         return cell
     }()
 
-    private lazy var sendAutocryptMessageCell: ActionCell = {
-        let cell = ActionCell()
+    private lazy var sendAutocryptMessageCell: UITableViewCell = {
+        let cell = UITableViewCell(style: .value1, reuseIdentifier: nil)
```
Comment (zeitschlag) [inline deltachat-ios/Controller/Settings/AdvancedViewController.swift:60]: What's the reason vor `.value1` here? From the docs:

> A style for a cell with a label on the left side of the cell with left-aligned and black text; _on the right side is a label that has smaller blue text and is right-aligned._

Do we need that _smaller blue text_ or would `default` suffice, also for `manageKeysCell`?

File: deltachat-ios/Controller/Settings/AdvancedViewController.swift [hunk 2]
```diff
@@ -219,20 +219,32 @@ internal final class AdvancedViewController: UITableViewController, ProgressAler
...
-        let serverSection = SectionConfigs(
-            headerTitle: String.localized("pref_server"),
-            footerTitle: String.localized("pref_only_fetch_mvbox_explain"),
-            cells: [accountSettingsCell, sentboxWatchCell, sendCopyToSelfCell, mvboxMoveCell, onlyFetchMvboxCell])
-        return [viewLogSection, experimentalSection, appAccessSection, autocryptSection, serverSection]
+
+        if dcContext.isChatmail {
+            let encryptionSection = SectionConfigs(
+                headerTitle: String.localized("pref_encryption"),
+                footerTitle: nil,
```
Comment (zeitschlag) [inline deltachat-ios/Controller/Settings/AdvancedViewController.swift:226]: We could think about adding a custom `init` for `SectionConfigs` with `nil` as default value for `footerTitle` 🤔

**Review by zeitschlag — APPROVED** [review]
LGTM. From what I understand, this requires to update the core-dependency, right?

Oh, and I just saw this message in the Xcode console:

> 🧡 [1] deltachat-ffi/src/lib.rs:285: dc_get_config(): invalid key 'is_chatmail'

This probably says that I didn't update core, right?

File: deltachat-ios/Controller/Settings/AdvancedViewController.swift (hunk 2 above)
Comment (r10s) [reply deltachat-ios/Controller/Settings/AdvancedViewController.swift:226]: yes, maybe that makes sense. however, the pattern is used also in many other views, so, i'll stay with it for now, it is easy enough

**Comment by r10s** [comment]
> This probably says that I didn't update core, right?

core was not updated on the `main`, will do that now

**Pull request merged**